includes/class-newspack-popups-settings.php

```diff
@@ -60,6 +60,99 @@ public static function set_settings( $options ) {
 		}
 	}
 
+	/**
+	 * Update settings from a specific section.
+	 *
+	 * @param WP_Request $request Request object with section and settings to update.
+	 *
+	 * @return array|WP_Error The settings list or error if a setting update fails.
+	 */
+	public static function update_section( $request ) {
+		$section  = $request['section'];
+		$settings = $request['settings'];
+		foreach ( $settings as $key => $value ) {
+			$updated = self::update_setting( $section, $key, $value );
+			if ( is_wp_error( $updated ) ) {
+				return $updated;
+			}
+		}
+		return self::get_settings( true );
+	}
+
+	/**
+	 * Update a setting from a provided section.
+	 *
+	 * @param string $section The section to update.
+	 * @param string $key     The key to update.
+	 * @param mixed  $value   The value to update.
+	 *
+	 * @return bool|WP_Error Whether the value was updated or error if key does not match settings configuration.
+	 */
+	private static function update_setting( $section, $key, $value ) {
+		$config = self::get_setting_config( $section, $key );
+		if ( ! $config ) {
+			return new WP_Error( 'newspack_ads_invalid_setting_update', __( 'Invalid setting.', 'newspack-ads' ) );
+		}
+		if ( isset( $config['options'] ) && is_array( $config['options'] ) ) {
+			$accepted_values = array_map(
+				function ( $option ) {
+					return $option['value'];
+				},
+				$config['options']
+			);
+			if ( ! in_array( $value, $accepted_values, true ) ) {
+				// translators: %s is the description of the option.
+				return new WP_Error( 'newspack_ads_invalid_setting_update', sprintf( __( 'Invalid setting value for "%s".', 'newspack-ads' ), $config['description'] ) );
+			}
+		}
+		$updated = update_option( $config['key'], self::sanitize_setting_option( $config['type'], $value ) );
+		return $updated;
+	}
+
+	/**
+	 * Retrieves a sanitized setting value to be stored as wp_option.
+	 *
+	 * @param string $type The type of the setting.
+	 * @param mixed  $value The value to sanitize.
+	 *
+	 * @return mixed The sanitized value.
+	 */
+	private static function sanitize_setting_option( $type, $value ) {
+		switch ( $type ) {
+			case 'int':
+			case 'integer':
+			case 'boolean':
+				return (int) $value;
+			case 'float':
+				return (float) $value;
+			case 'string':
+				return sanitize_text_field( $value );
+			case 'select':
```

Comment:
Nit: since `select` is determined by the existence of `options` and not by its type, should we keep this case?

Reply:
Guess it's not necessary—removed in ab48f5a.